model/histogram/float_histogram.go

```diff
@@ -824,55 +825,83 @@ func (i *floatBucketIterator) Next() bool {
 		return false
 	}
 
-	// Copy all of these into local variables so that we can forward to the
-	// next bucket and then roll back if needed.
-	origIdx, spansIdx, idxInSpan := i.origIdx, i.spansIdx, i.idxInSpan
-	span := i.spans[spansIdx]
-	firstPass := true
-	i.currCount = 0
-
```

Comment:
Moved into the `else` branch for the uncommon case: https://github.com/prometheus/prometheus/pull/13212/files#diff-e8eb6f5081357b38dc1b6f160d8fbf173a0ced8df4f772df67bdc7a7ab0fe9d6R853-R860